framework_tvb/tvb/core/adapters/abcuploader.py

```diff
@@ -77,8 +95,55 @@ def _prelaunch(self, operation, uid=None, available_disk_space=0, view_model=Non
         self.meta_data.update({DataTypeMetaData.KEY_SUBJECT: view_model.data_subject})
         self.generic_attributes.subject = view_model.data_subject
 
+        trait_upload_field_names = list(self.get_form_class().get_upload_information().keys())
+        for upload_field_name in trait_upload_field_names:
+            self._decrypt_content(view_model, upload_field_name)
+
         return ABCSynchronous._prelaunch(self, operation, uid, available_disk_space, view_model, **kwargs)
 
+    @staticmethod
+    def _decrypt_content(view_model, trait_upload_field_name):
+        if view_model.encrypted_aes_key is None:
```

Comment:
Maybe move this check before the loop over `trait_upload_field_names ` as it applies to all inputs ?

Reply:
Okay.